scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -74,27 +69,6 @@ abstract class ScanResultsStorage : PackageBasedScanStorage {
         return result
     }
 
-    /**
-     * Return all [ScanResult]s contained in this [ScanResultsStorage] corresponding to the given [packages] that
-     * are [compatible][ScannerMatcher.matches] with the provided [scannerMatcher] wrapped in a [Result]. Also,
-     * [Package.sourceArtifact], [Package.vcs], and [Package.vcsProcessed] are used to check if the scan result matches
-     * the expected source code location. That check is important to find the correct results when different revisions
-     * of a package using the same version name are used (e.g. multiple scans of a "1.0-SNAPSHOT" version during
-     * development).
-     */
-    fun read(packages: Collection<Package>, scannerMatcher: ScannerMatcher): Result<Map<Identifier, List<ScanResult>>> {
```

Comment:
Any comment on *why* this function is actually unused? I mean, as we're pretty much always interested in results of multiple packages, it would seem more efficient to *only* use this function in order to not apply the scanner matcher for every package we request.

Reply:
I initially did not use it in the new scanner implementation for simplicity. Also, the new scanner converts package scan results into provenance scan results and reuses them if another package has the same provenance. And since the switch to the new scanner two years ago the downloading of stored scan results never seemed to be a performance problem so no one looked into optimizing the performance, e.g. by taking this function into use again.

Reply:
Thanks for the details. I'll leave it up to you whether you think it's worth putting some of that into the commit message.

Reply:
I have added some of it to the commit message.

Reply:
Just FYI @mnonnenmacher, we might have a use-case to add back the API for reading multiple packages at once: Is there is a general problem with some scanner backend, there is no point in continuing reading from it if the first read already fails, and we'd like to be able to fail all of the reads at once.

Reply:
Taking it back into use might require a bigger refactoring of the `Scanner`, losing some of the performance optimizations along the way, e.g. not downloading scan results multiple times for packages that share the same provenance.
If the goal is to disable a storage if a certain error occurred, that could be implemented also without introducing such a function.

Reply:
> that could be implemented also without introducing such a function.

What do you have in mind @mnonnenmacher? Like a special `Result` value / exception being throw that signals the caller that no further packages should be tried?

Reply:
Yes, something like that. I think the tricky part would be to decide which kind of error should trigger this behavior. In most cases it might be better to fix the storage than to rescan everything, but that depends on the specific setup.

Reply:
> I think the tricky part would be to decide which kind of error should trigger this behavior.

I believe this should simply be signaled via a new special exception that implementations may throw...